File: Qt/view/TithiHolder.cpp
#include "TithiHolder.h"
#include "ui_TithiHolder.h"
#include "../util/Event.h"

TithiHolder::TithiHolder()
: ui(new Ui::TithiHolder)
{
	ui->setupUi(this);
}

TithiHolder::~TithiHolder()
{
	delete ui;
}

void TithiHolder::fillLayout()
{
	Event* e = activeEvent();
	if (e) {
		ui->mDegree->setText(QString::number(e->getDegree()));
		ui->mText0->setText(e->normalizedRangeString());
		setColorByEventMode(ui->mText0, e);
	} 
	else {
		ui->mDegree->setText("");
		ui->mText0->setText("");
	}
}

File: Astromaximum-eclipse/src/com/astromaximum/android/view/TithiHolder.java
package com.astromaximum.android.view;

import com.astromaximum.android.R;
import com.astromaximum.android.util.Event;

public class TithiHolder extends ViewHolder {

	public TithiHolder(SummaryItem si) {
		super(si, R.layout.item_tithi, LAYOUT_FLAG_TEXT0 | LAYOUT_FLAG_DEGREE);
	}

	@Override
	public void fillLayout() {
		Event e = getActiveEvent();
		if (e != null) {
			mDegree.setText(Integer.toString(e.getDegree()));
			mText0.setText(e.normalizedRangeString());
			setColorByEventMode(mText0, e);
		} else {
			mDegree.setText("");
			mText0.setText("");
		}
	}
}
